app/controller_services/shopping_list_items_controller/create_service.rb

```diff
@@ -57,8 +59,28 @@ def aggregate_list
       @aggregate_list ||= shopping_list.aggregate_list
     end
 
+    def game
+      @game ||= aggregate_list.game
+    end
+
+    def aggregate_list_item
+      @aggregate_list_item ||= aggregate_list.list_items.find_by('description ILIKE ?', params[:description])
+    end
+
     def all_matching_list_items
-      aggregate_list.game.shopping_list_items.where('description ILIKE ?', params[:description])
+      @all_matching_list_items ||= game.shopping_list_items.where(
+        'description ILIKE ?', params[:description],
+      )
+    end
+
+    def lists_to_be_changed
+      list_ids = if params[:unit_weight] && params[:unit_weight] != aggregate_list_item&.unit_weight
+                   all_matching_list_items.pluck(:list_id).push(shopping_list.id)
```

Comment:
The list IDs will already include `shopping_list.id` if there is a matching item already on that shopping list. However, there's no real disadvantage to having it on the list twice and it would be more code to check or call `uniq` than to just leave it.